build.gradle

```diff
@@ -155,11 +154,13 @@ SourceSet setupExtraSourceSet(SourceSet baseSourceSet, String extra) {
     extraSourceSet.resources.srcDirs = ["src/${extra}/${name}/resources"]
     extraSourceSet.compileClasspath += project.sourceSets.api.output
     extraSourceSet.compileClasspath += project.sourceSets.main.output
+    extraSourceSet.runtimeClasspath += project.sourceSets.main.runtimeClasspath
     if (baseSourceSet != project.sourceSets.main) {
         //If the base sourceSet is main it already is the extra source set and has a reference to the base one from before this if statement
         extraSourceSet.compileClasspath += getExtraSourceSet(project.sourceSets.main, extra).get().output
         extraSourceSet.compileClasspath += baseSourceSet.output
-        extraSourceSet.runs.modIdentifier = name
+        extraSourceSet.runtimeClasspath += baseSourceSet.runtimeClasspath
+        neoForge.mods.maybeCreate(name).modSourceSets.add(extraSourceSet)
```

Comment:
I believe this will end up causing the datagen sourcesets to be with the corresponding module's mod, when it should only load for datagen instead of for normal runs as well.